### DISCUSSION TITLE
USB device Deny in Client Machine

### DISCUSSION THREAD

**Discussion by perlland** [ltsp/ltsp#565] [General]
Hi,
Could you please suggest how to deny USB device in ltsp client machine. Can it be configured in ltsp.conf file or can be configured in client image???
Earlier suggestion is deeply appreciated.
Thanking you

**Comment by alkisg** [comment]
Hi, this question isn't related to LTSP.
You may google for "how to deny USB devices in Linux".
When you have an answer that you tested it in non-LTSP environment and works the way you want it, post it here.

Then you can either apply it inside the LTSP image in the same way that you implemented it in the non-LTSP environment,
or you can apply it dynamically from ltsp.conf when the client boots, with a POST_INIT_x command.

**Reply by perlland** [reply]
I have tried with USBGuard in Ubuntu  and it works perfectly fine over there but when I implement it in LTSP image, Daemon failed for USBGuard package in client machine.
Udev Daemon is not working in LTSP client machine.

**Comment by alkisg** [comment]
Udev is certainly working in all LTSP clients, otherwise newly inserted devices wouldn't work at all.
For example:

```$ ps faux | grep udev
root         408  0.0  0.0  24164  7624 ?        Ss   Οκτ04   0:09 /lib/systemd/systemd-udevd
```

LTSP doesn't touch udev at all unless the user adds MULTISEAT=1 or UDEV_SEAT_x parameters in ltsp.conf. It's not the one to blame here.

How are you creating your LTSP image, maybe you lack necessary packages for USBGuard to work? Try to write the same image locally in a disk and boot it without LTSP, to see whether the problem is in the image.
